packages/react-sdk/src/contexts/XMTPContext.tsx

```diff
@@ -62,6 +129,93 @@ export const XMTPProvider: React.FC<React.PropsWithChildren> = ({
   const [error, setError] = useState<unknown | null>(null);
   const initializingRef = useRef(false);
 
+  // combine all processors into a single object
+  const processors = useMemo(() => {
+    const finalCacheConfig = [...defaultCacheConfig, ...(cacheConfig ?? [])];
+    return {
```

Comment:
do you want to add some unit tests for this logic? a bit hard to read, esp with the nested reduce.

Reply:
agreed! i basically ran out of time for testing and had to pick and choose my battles for a release. i can do this in a fast-follow.